### DISCUSSION TITLE
When clicking the about option in the sidebar then open the about page in the next js using typescript.

### DISCUSSION THREAD

**Discussion by MinalPatilRPA** [tailwindlabs/tailwindcss#12809] [Help]
When clicking the about option in the sidebar then open the about page in the next js using typescript. 

Note: 1. The About page is only displayed in the hero section.
          2. Only display a particular page. 

How can I solve this code?

![image](https://github.com/tailwindlabs/tailwindcss/assets/122990718/bc717a94-f6b0-49d4-abf0-bab4d474b0e9)

Refer above image. The blue section shows the sidebar, the red section shows the nav bar and the green section shows the hero page. 

Mycode: -

about.tsx: 
import React from 'react'

export const About = () => {
  return (
    <div className='bg-[#131222] pb-12 pt-14 md:pt-32' id='about'>
      <div className='grid grid-col-1 md:grid-cols-2 w-4/5 mx-auto gap-12 items-center'>
          <div>
            <h1 className='text-xl font-bold text-white uppercase md-[1rem]'>About ME</h1>
            <h2></h2>
          </div>
          <div></div>
      </div>
    </div>
  )
}

sidebar.tsx:

import React,{useState} from 'react'
import Link  from 'next/link'
import Image from "next/image"
import { Bars3Icon } from '@heroicons/react/20/solid'
export const Sidebar = () => {

  const [open, setOpen] = useState(true);
 return (
  < >
        <div className='flex items-center justify-end p-4'>
                <div className='md:hidden text-white font-bold pr-20'>
                    <h1 className=' text-xl md:text-2xl '> SOFTWARE <span className='text-yellow-600'>DEV</span>
                    </h1> 
                </div>
                <Bars3Icon className='relative cursor-pointer text-yellow-400 w-7' onClick={() => setOpen(!open)}/>
        </div>
        
        <div className='mt-4 md:mt-6 '>
            {/* DP */}
          <div className='md:w-[150px] w-[70px] h-[70px] border-8 border-gray-500 relative lg:flex items-center rounded-full md:h-[150px] mx-auto '>
              <Image src="/images/u1.jpg" alt="user" layout='fill' className='object-cover rounded-full ' />                  
          </div>

            {/* Name */}
          <div className='mt-2 text-white text-2xl md:text-3xl font-bold items-center  mx-auto flex justify-center'><h2>Minal Patil</h2></div>

        {/* Social links */}
           <div className='mt-3 mx-auto items-center flex justify-center gap-3'>
                <Link href={"/"} className='bg-gray-600 rounded-full w-8 h-8 md:w-9 md:h-9 items-center flex justify-center'>           
                    <i className='bx bxl-github text-xl text-white'></i>                                       
                </Link> 

                <Link href={"/"} className='bg-gray-600 rounded-full w-8 h-8 md:w-9 md:h-9 items-center flex justify-center '>           
                    <i className='bx bxl-twitter text-xl text-white'></i>                                        
                </Link> 

                <Link href={"/"} className='bg-gray-600 rounded-full w-8 h-8 md:w-9 md:h-9 items-center flex justify-center'>           
                    <i className='bx bxl-instagram text-xl text-white'></i>                                        
                </Link> 

                <Link href={"/"} className='bg-gray-600 rounded-full w-8 h-8 md:w-9 md:h-9 items-center flex justify-center'>           
                    <i className='bx bxl-skype text-xl text-white'></i>                                   
                </Link> 

                <Link href={"/"} className='bg-gray-600 rounded-full w-9 h-9 items-center flex justify-center'>           
                    <i className='bx bxl-linkedin text-xl text-white'></i>                                       
                </Link> 
          </div> 

          <div className='m-12 md:mt-20 flex flex-col px-2 md:px-8  gap-5 relative text-[#a8a9b4] font-semibold text-xl  '>
                {/* <Link href={"/"} className=''>           
                    <i className='bx bx-home custom-icon'></i>             
                    <span className='pl-6'>Home</span>         
                </Link>  */}

                <Link href={"/about"}  >
                    <i className='bx bx-user custom-icon'></i>
                    <span className='pl-6'>About</span>
                </Link> 

                 <Link href={"/resume"} >
                    <i className='bx bx-file-blank'></i>
                    <span className='pl-6'>Resume</span>
                </Link> 

                <Link href={"/service"}>
                    <i className='bx bx-server'></i>
                    <span className='pl-6'>Service</span>
                </Link> 
                
                 <Link href={"/skill"}>
                    <i className='bx bx-objects-horizontal-left'></i>
                    <span className='pl-6'>skill</span>
                </Link>  

                <Link href={"/projects"}>
                    <i className='bx bx-book-content'></i>
                    <span className='pl-6'>Projects</span>
                </Link> 

                <Link href={"/contact"}>
                    <i className='bx bx-mobile'></i>
                    <span className='pl-6'>Contact</span>
                </Link>  
          </div>
        </div>
     
  </>
  )
}

**Comment by makibat** [comment]
Hey!

I'm not sure I fully understand your question, but this sounds more like a layout issue on your end. Using the [app router](https://nextjs.org/docs/app), the structure should be as follows:

```
├── app
│   ├── about
│   │   └── page.tsx
│   ├── resume
│   │   └── page.tsx
│   ├── service
│   │   └── page.tsx
│   ...
│   ├── page.tsx
│   └── layout.tsx
└── ...
```

And then you could import your  `Navbar` and `Sidebar` components in `layout.tsx` and that layout will apply to all routes:
``` js

interface RootLayoutProps {
  children: React.ReactNode;
}

export default function RootLayout({
  children
}: RootLayoutProps) {
  return (
    <html lang='en'>
      <body>
        <Sidebar />
        <div>
          <Navbar />
          <Hero>{children}</Hero>
        </div>
      </body>
    </html>
  );
}
```

If you don't want to apply the same layout to all routes, I suggest you look into [route groups](https://nextjs.org/docs/app/building-your-application/routing/route-groups)

Hope that helps!
